packages/x-tree-view/src/internals/models/treeView.ts

```diff
@@ -38,4 +38,5 @@ export type TreeViewPublicAPI<
 
 export type TreeViewExperimentalFeatures<
   TSignatures extends readonly TreeViewAnyPluginSignature[],
-> = { [key in MergeSignaturesProperty<TSignatures, 'experimentalFeatures'>]?: boolean };
+  TOptionalSignatures extends readonly TreeViewAnyPluginSignature[] = [],
+> = MergeSignaturesProperty<[...TSignatures, ...TOptionalSignatures], 'experimentalFeatures'>;
```

Comment:
I also added support for the optional plugins even if we never use it for now, just to be consistent with the `instance` and `params`